<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>StarSignDNA &#8212; Cucumber 1.0.3 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=baaebd52"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="installation.html" />
    <link rel="prev" title="Welcome to Cucumber’s documentation!" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="starsigndna">
<h1>StarSignDNA<a class="headerlink" href="#starsigndna" title="Link to this heading">¶</a></h1>
<p>StarSignDNA algorithm for mutational signature analysis which offers efficient refitting and de novo
mutational signature extraction. StarSignDNA is capable of deciphering well-differentiated signatures linked to known
mutagenic mechanisms and demonstrates strong associations with patient clinical features. The package offers a user-
friendly interface and data visualization routines.</p>
<section id="mutation-signature-analysis-package">
<h2>Mutation signature analysis package<a class="headerlink" href="#mutation-signature-analysis-package" title="Link to this heading">¶</a></h2>
<a class="reference external image-reference" href="https://opensource.org/licenses/MIT"><img alt="MIT License" src="https://img.shields.io/badge/license-MIT-blue.svg" />
</a>
<ul class="simple">
<li><p>Free software: <a class="reference external" href="https://opensource.org/licenses/MIT">MIT license</a></p></li>
<li><p>Documentation: Link to be provided</p></li>
</ul>
</section>
<section id="paper">
<h2>Paper<a class="headerlink" href="#paper" title="Link to this heading">¶</a></h2>
<p>Preprint link <a class="reference external" href="https://www.biorxiv.org/content/10.1101/2024.06.29.601345v1">https://www.biorxiv.org/content/10.1101/2024.06.29.601345v1</a></p>
</section>
<section id="features">
<h2>Features<a class="headerlink" href="#features" title="Link to this heading">¶</a></h2>
</section>
<section id="stable-release">
<h2>Stable release<a class="headerlink" href="#stable-release" title="Link to this heading">¶</a></h2>
<p>To install StarSign,</p>
<p>you can install the package directly from PyPi:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>   <span class="n">pip</span> <span class="n">install</span> <span class="n">starsigndna</span>

<span class="n">Alternatively</span><span class="p">,</span>  <span class="n">you</span> <span class="n">can</span> <span class="n">install</span> <span class="n">it</span> <span class="n">via</span> <span class="n">terminal</span> <span class="n">by</span> <span class="n">running</span> <span class="n">this</span> <span class="n">command</span><span class="p">::</span>

   <span class="mf">1.</span> <span class="n">Download</span> <span class="n">StarSign</span> <span class="kn">from</span> <span class="nn">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">uio</span><span class="o">-</span><span class="n">bmi</span><span class="o">/</span><span class="n">StarSignDNA</span>
   <span class="mf">2.</span> <span class="n">Unzip</span> <span class="n">StarSignDNA</span><span class="o">-</span><span class="n">master</span><span class="o">.</span><span class="n">zip</span>
   <span class="mf">3.</span> <span class="n">cd</span> <span class="n">StarSigndna</span><span class="o">-</span><span class="n">master</span><span class="o">/</span>
   <span class="mf">4.</span> <span class="n">pip</span> <span class="n">install</span> <span class="o">-</span><span class="n">e</span> <span class="o">.</span>
</pre></div>
</div>
</section>
<section id="getting-started">
<h2>Getting started<a class="headerlink" href="#getting-started" title="Link to this heading">¶</a></h2>
<p>To obtain help:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">starsigndna</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<p>Usage: starsigndna [OPTIONS] COMMAND [ARGS]…</p>
</section>
<section id="options">
<h2>Options<a class="headerlink" href="#options" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p><strong>–install-completion</strong>: Install completion for the current shell.</p></li>
<li><p><strong>–show-completion</strong>: Show completion for the current shell, to copy it or customize the installation.</p></li>
<li><p><strong>–help</strong>: Show this message and exit.</p></li>
</ul>
</section>
<section id="commands">
<h2>Commands<a class="headerlink" href="#commands" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p><strong>count-mutation</strong>: Count mutation types in VCF file.</p></li>
<li><p><strong>denovo</strong>: Performs denovo Mutational Signatures analysis.</p></li>
<li><p><strong>refit</strong>: Mutational Signatures Refit Parameters</p></li>
</ul>
</section>
<section id="running-mutational-signature-refit-algorithm">
<h2>Running mutational signature refit algorithm<a class="headerlink" href="#running-mutational-signature-refit-algorithm" title="Link to this heading">¶</a></h2>
<p>The refitting algorithm takes as input a mutational catalog and a COSMIC mutational signature file. The user can also specify signatures to be considered instead of using the full COSMIC matrix or a subset matrix:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">starsigndna</span> <span class="n">refit</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
</section>
<section id="starsigndna-refit-parameters">
<h2>StarSignDNA Refit Parameters<a class="headerlink" href="#starsigndna-refit-parameters" title="Link to this heading">¶</a></h2>
<p><strong>Arguments</strong></p>
<ul class="simple">
<li><p><strong>matrix_file</strong> (TEXT): Tab separated matrix file [default: None] [required]</p></li>
<li><p><strong>signature_file</strong> (TEXT): Tab separated matrix file [default: None] [required]</p></li>
</ul>
<p><strong>Options</strong></p>
<ul class="simple">
<li><p><strong>–ref_genome</strong> (TEXT): Path to the reference genome [default: None]</p></li>
<li><p><strong>–n_bootstraps</strong> (INTEGER): Number of bootstraps [default: 200]</p></li>
<li><p><strong>–opportunity_file</strong> (TEXT): Path to the opportunity file [default: None]</p></li>
<li><p><strong>–numeric_chromosomes</strong>: If set, chromosome names are numeric [default: no-numeric-chromosomes]</p></li>
<li><p><strong>–no_numeric_chromosomes</strong>: If set, chromosome names are not numeric [default: no-numeric-chromosomes]</p></li>
<li><p><strong>–genotyped</strong>: If set, VCF file has genotype information for many samples [default: genotyped]</p></li>
<li><p><strong>–no_genotyped</strong>: If set, VCF file does not have genotype information for many samples [default: genotyped]</p></li>
<li><p><strong>–output_folder</strong> (TEXT): Path to the output folder [default: output/]</p></li>
<li><p><strong>–signature_names</strong> (TEXT): Comma separated list of signature names [default: None]</p></li>
<li><p><strong>–n_iterations</strong> (INTEGER): Number of iterations [default: 1000]</p></li>
<li><p><strong>–help</strong>: Show this message and exit</p></li>
</ul>
<p>Running StarSignDNA refit:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">starsigndna</span> <span class="n">refit</span> <span class="n">example_data</span><span class="o">/</span><span class="n">M_catalogue</span><span class="o">.</span><span class="n">txt</span> <span class="n">example_data</span><span class="o">/</span><span class="n">COSMICv34</span><span class="o">.</span><span class="n">txt</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">folder</span> <span class="o">/</span><span class="n">test_result</span> <span class="o">--</span><span class="n">signature</span><span class="o">-</span><span class="n">names</span> <span class="n">SBS40c</span><span class="p">,</span><span class="n">SBS2</span><span class="p">,</span><span class="n">SBS94</span>
<span class="n">starsigndna</span> <span class="n">refit</span> <span class="n">example_data</span><span class="o">/</span><span class="n">tcga_coad_single</span><span class="o">.</span><span class="n">vcf</span> <span class="n">example_data</span><span class="o">/</span><span class="n">sig_cosmic_v3_2019</span><span class="o">.</span><span class="n">txt</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">folder</span> <span class="o">/</span><span class="n">output</span> <span class="o">--</span><span class="n">signature</span><span class="o">-</span><span class="n">names</span> <span class="n">SBS40c</span><span class="p">,</span><span class="n">SBS2</span><span class="p">,</span><span class="n">SBS94</span> <span class="o">--</span><span class="n">ref</span><span class="o">-</span><span class="n">genome</span>
</pre></div>
</div>
<p>When the –signature-names option is used, the default number of signature is 3, but we recommend minimum of 5 signatures.
The test data is provided in the example_data folder. To convert <a href="#id1"><span class="problematic" id="id2">*</span></a>.vcf to a matrix, the user must provide the path to the reference genome using the option –ref-genome.</p>
<p>The user can also provide the distribution of triplets in a reference genome/exome or normal tissue in the same patient (Opportunity matrix) using the option –opportunity-file human-genome/human-exome.</p>
</section>
<section id="running-mutational-signature-de-novo-algorithm">
<h2>Running mutational signature de novo algorithm<a class="headerlink" href="#running-mutational-signature-de-novo-algorithm" title="Link to this heading">¶</a></h2>
<p>The de novo algorithm takes as input a mutational catalog and infers the exposure matrix and mutational signature matrix. The COSMIC mutational signature file is provided to compute the cosine similarity:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">starsigndna</span> <span class="n">denovo</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<section id="performs-denovo-mutational-signatures-analysis">
<h3>Performs denovo Mutational Signatures analysis<a class="headerlink" href="#performs-denovo-mutational-signatures-analysis" title="Link to this heading">¶</a></h3>
<p><strong>Arguments</strong></p>
<ul class="simple">
<li><p><strong>matrix_file</strong> (TEXT): Tab separated matrix file [default: None] [required]</p></li>
<li><p><strong>n_signatures</strong> (INTEGER): Number of signatures to identify [default: None] [required]</p></li>
</ul>
<p><strong>Options</strong></p>
<ul class="simple">
<li><p><strong>–lambd</strong> (FLOAT): Regularization parameter [default: 0.7]</p></li>
<li><p><strong>–opportunity-file</strong> (TEXT): The distribution of triplets in a reference ‘human-genome’ or ‘human-exome’ or normal tissue [default: None]</p></li>
<li><p><strong>–cosmic-file</strong> (TEXT): Tab separated cosmic file [default: None]</p></li>
<li><p><strong>–numeric-chromosomes</strong>: If set, chromosome names are numeric [default: no-numeric-chromosomes]</p></li>
<li><p><strong>–no-numeric-chromosomes</strong>: If set, chromosome names are not numeric [default: no-numeric-chromosomes]</p></li>
<li><p><strong>–genotyped</strong>: If set, VCF file has genotype information for many samples [default: genotyped]</p></li>
<li><p><strong>–no-genotyped</strong>: If set, VCF file does not have genotype information for many samples [default: genotyped]</p></li>
<li><p><strong>–max-em-iterations</strong> (INTEGER): Maximum EM iterations [default: 100]</p></li>
<li><p><strong>–max-gd-iterations</strong> (INTEGER): Maximum GD iterations [default: 50]</p></li>
<li><p><strong>–file-extension</strong> (TEXT): File extension [default: None]</p></li>
<li><p><strong>–ref-genome</strong> (TEXT): Path to the reference genome [default: None]</p></li>
<li><p><strong>–output-folder</strong> (TEXT): Path to the output folder [default: output/]</p></li>
<li><p><strong>–help</strong>: Show this message and exit</p></li>
</ul>
<p>Step 1: Grid Search: The grid uses cross-validation to find the optimal pairwise (k and λ) by going to the snakemake folder and opening the running file (Snakefile) to check all the paths and input files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="n">snakemake</span>
<span class="n">vi</span> <span class="n">Snakefile</span>
</pre></div>
</div>
<p>Step 2: In the Snakefile, provide the range of the number of signatures k and λ for the grid search to determine the optimal k and λ:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">localrules</span><span class="p">:</span> <span class="nb">all</span>
<span class="n">ks</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">10</span><span class="p">)):</span> <span class="n">default</span> <span class="nb">range</span> <span class="n">of</span> <span class="n">the</span> <span class="n">number</span> <span class="n">of</span> <span class="n">signatures</span>
<span class="n">lambdas</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">]:</span> <span class="n">default</span> <span class="nb">range</span> <span class="n">of</span> <span class="n">λ</span>
</pre></div>
</div>
<p>Input mutational catalogue needs to be provided in the dataset folder:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">test_train_split</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span> <span class="s2">&quot;results/</span><span class="si">{dataset}</span><span class="s2">/M_catalogue.txt&quot;</span>
</pre></div>
</div>
<p>Running the grid search:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span> <span class="o">-</span><span class="n">j</span> <span class="n">num_cpu</span>
</pre></div>
</div>
<p>To check manually the optimal k and λ from the output:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sort</span> <span class="o">-</span><span class="n">k3n</span><span class="p">,</span><span class="mi">3</span> <span class="n">results</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="nb">all</span><span class="o">.</span><span class="n">csv</span>
</pre></div>
</div>
<p>Run denovo using optimal k=4 and λ=0.1:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">starsigndna</span> <span class="n">denovo</span> <span class="n">snakemake</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">M_catalogue</span><span class="o">.</span><span class="n">txt</span> <span class="mi">4</span> <span class="mf">0.1</span> <span class="o">--</span><span class="n">cosmic</span><span class="o">-</span><span class="n">file</span> <span class="n">example_data</span><span class="o">/</span><span class="n">COSMICv34</span><span class="o">.</span><span class="n">txt</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">folder</span> <span class="o">/</span><span class="n">test_result</span>
</pre></div>
</div>
</section>
</section>
<section id="data">
<h2>Data<a class="headerlink" href="#data" title="Link to this heading">¶</a></h2>
<p>Simulated data publicly available is available on FigShare: <a class="reference external" href="https://doi.org/10.6084/m9.figshare.28105610">https://doi.org/10.6084/m9.figshare.28105610</a></p>
<ul class="simple">
<li><p><strong>–flatness_100mutations folder</strong>: Table 1. dataset with around 100 mutations per sample</p></li>
<li><p><strong>–flatness_500mutations folder</strong>: Table 1. dataset with around 500 mutations per sample</p></li>
<li><p><strong>–flatnessdownsample_50_mutations folder</strong>: Table 1. dataset with around 50 mutations per samples</p></li>
<li><p><strong>–flatnessdownsample_70_mutations folder</strong>: Table 1. dataset with around 70 mutations per samples</p></li>
<li><p><strong>–flatnessdownsample_80_mutations folder</strong>: Table 1. dataset with around 80 mutations per samples</p></li>
<li><p><strong>–noflatness_100mutations folder</strong>: Table 2. dataset with around 100 mutations per sample</p></li>
<li><p><strong>–noflatness_500mutations folder</strong>: Table 2. dataset with around 500 mutations per sample</p></li>
</ul>
</section>
<section id="contact">
<h2>Contact<a class="headerlink" href="#contact" title="Link to this heading">¶</a></h2>
<p>Maintainer Name - <a class="reference external" href="mailto:chrisbop&#37;&#52;&#48;uio&#46;no">chrisbop<span>&#64;</span>uio<span>&#46;</span>no</a> or <a class="reference external" href="mailto:christianbope&#37;&#52;&#48;gmail&#46;com">christianbope<span>&#64;</span>gmail<span>&#46;</span>com</a></p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">Cucumber</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">StarSignDNA</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#mutation-signature-analysis-package">Mutation signature analysis package</a></li>
<li class="toctree-l2"><a class="reference internal" href="#paper">Paper</a></li>
<li class="toctree-l2"><a class="reference internal" href="#features">Features</a></li>
<li class="toctree-l2"><a class="reference internal" href="#stable-release">Stable release</a></li>
<li class="toctree-l2"><a class="reference internal" href="#getting-started">Getting started</a></li>
<li class="toctree-l2"><a class="reference internal" href="#options">Options</a></li>
<li class="toctree-l2"><a class="reference internal" href="#commands">Commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-mutational-signature-refit-algorithm">Running mutational signature refit algorithm</a></li>
<li class="toctree-l2"><a class="reference internal" href="#starsigndna-refit-parameters">StarSignDNA Refit Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-mutational-signature-de-novo-algorithm">Running mutational signature de novo algorithm</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data">Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#contact">Contact</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Welcome to Cucumber’s documentation!</a></li>
      <li>Next: <a href="installation.html" title="next chapter">Installation</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2023, Christian Bope.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/readme.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>